classes/WP_CLI/Action/Delete.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/**
+ * WP-CLI command: action-scheduler action delete
+ */
+class ActionScheduler_WPCLI_Action_Delete_Command extends ActionScheduler_WPCLI_Command {
+
+	/** @var int[] */
+	protected $action_ids = array();
+
+	/** @var array<string, int> */
+	protected $action_counts = array(
+		'deleted' => 0,
+		'total'   => 0,
+	);
+
+	/**
+	 * Construct.
+	 *
+	 * @param string[]              $args       Positional arguments.
+	 * @param array<string, string> $assoc_args Keyed arguments.
+	 */
+	public function __construct( array $args, array $assoc_args ) {
+		parent::__construct( $args, $assoc_args );
+
+		$this->action_ids             = array_map( 'absint', $args );
+		$this->action_counts['total'] = count( $this->action_ids );
+
+		add_action( 'action_scheduler_deleted_action', array( $this, 'action__deleted' ) );
+	}
+
+	/**
+	 * Execute.
+	 *
+	 * @uses \ActionScheduler_Store::delete_action()
+	 * @uses \WP_CLI::warning()
+	 * @uses \WP_CLI::success()
+	 * @return void
+	 */
+	public function execute() {
+		$store = \ActionScheduler::store();
+
+		$progress_bar = \WP_CLI\Utils\make_progress_bar(
+			sprintf(
+				/* translators: %d: number of actions to be deleted */
+				_n( 'Deleting %d action', 'Deleting %d actions', $this->action_counts['total'], 'action-scheduler' ),
+				number_format_i18n( $this->action_counts['total'] )
+			),
+			$this->action_counts['total']
+		);
+
+		foreach ( $this->action_ids as $action_id ) {
+			$store->delete_action( $action_id );
```

Comment:
Seems like we not handling failures gracefully, passing one invalid ID crashes the whole command. I think we can wrap this in a try catch, and then display the actions that we were not able to delete:

```
> wp action-scheduler action delete 1561 163333
Deleting 2 actions  50 % [========================================================>                                                        ] 0:00 / 0:00Fatal error: Uncaught InvalidArgumentException: Unidentified action 163333: we were unable to delete this action. It may may have been deleted by another process. in /usr/src/public_html/wp-content/plugins/action-scheduler/classes/data-stores/ActionScheduler_DBStore.php:746
Stack trace:
#0 /usr/src/public_html/wp-content/plugins/action-scheduler/classes/WP_CLI/Action/Delete.php(53): ActionScheduler_DBStore->delete_action(163333)
#1 /usr/src/public_html/wp-content/plugins/action-scheduler/classes/WP_CLI/ActionScheduler_WPCLI_Action_Command.php(119): ActionScheduler_WPCLI_Action_Delete_Command->execute()
#2 [internal function]: ActionScheduler_WPCLI_Action_Command->delete(Array, Array)
#3 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Dispatcher/CommandFactory.php(100): call_user_func(Array, Array, Array)
#4 [internal function]: WP_CLI\Dispatcher\CommandFactory::WP_CLI\Dispatcher\{closure}(Array, Array)
#5 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Dispatcher/Subcommand.php(491): call_user_func(Objec in /usr/src/public_html/wp-content/plugins/action-scheduler/classes/data-stores/ActionScheduler_DBStore.php on line 746
Error: There has been a critical error on this website.Learn more about troubleshooting WordPress. There has been a critical error on this website.
```